sos/report/plugins/instructlab.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (C) 2024 Red Hat, Inc., Jose Castillo <jcastillo@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+
+
+class Instructlab(Plugin, IndependentPlugin):
+    """
+    This plugin is used to capture information about
+    Instructlab installations.
+    InstructLab is an open source project for enhancing
+    large language models (LLMs) used in generative a
+    rtificial intelligence (gen AI) applications.
+    Instructlab can run either as a container, or directly
+    outside a container.
+    """
+
+    short_desc = 'Instructlab'
+    plugin_name = 'instructlab'
+    profiles = ('ai',)
+    containers = ('instructlab', 'ilab',)
+    commands = ('ilab',)
+
+    option_list = [
+        PluginOpt('ilab_user', default='', val_type=str,
+                  desc='user that runs instructlab'),
+        PluginOpt('ilab_conf_dir', default='', val_type=str,
+                  desc='instructlab data directory'),
+        PluginOpt('get-cache', default=False,
+                  desc='Capture models and osci cached data')
+    ]
+
+    def setup(self):
+        cont_share_conf_path = "/usr/share/instructlab/config"
+        cont_opt_path = "/opt/app-root/src"
+        # .cache dir contains the models and oci directories
+        # which can be quite big. We'll gather this only if
+        # specifying it via command line option
+        cache_dir = "/.cache/instructlab"
+        # .config is where the configuration yaml files can
+        # be found. We gather this always.
+        config_dir = "/.config/instructlab"
+        # In the .local directory we can find datasets,
+        # chat logs, taxonomies, and other very useful data
+        # We gather this always.
+        local_share_dir = "/.local/share/instructlab"
+
+        # container paths
+        cont_cache_path = f"{cont_opt_path}{cache_dir}"
+        cont_config_path = f"{cont_opt_path}{config_dir}"
+        cont_local_path = f"{cont_opt_path}{local_share_dir}"
```

Comment:
I would prefer using `os.path.join` method to joining.. well.. paths :)

Reply:
ack